File: Assignment3/program4.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program4.c
//  Description : Reads one character and converts it to lower and upper case
//  Author      : Akhilesh.P.Sonavane
//  Date        : 10/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////
// 
//  Function Name : DisplayConvert
//  Description   : Accept input character & coverts it into upper or lower case
//  Input         : Character
//  Output        : Character
//
////////////////////////////////////////////////////////////////

void DisplayConvert(char cVal)
{
    if(cVal >= 'a' && cVal <= 'z')
    {
        printf("%c\n",cVal - 32);
    }
        
    else if(cVal >= 'A' && cVal <= 'Z')
    {
        printf("%c\n",cVal + 32);
    }
}

int main()
{
    char cValue = '\0';

    printf("Enter Character\n");
    scanf("%c",&cValue);

    DisplayConvert(cValue);
    return 0;
}

File: Assignment23/program4.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program4.java
//  Description : Accept a character character array coverting it into String and checks if its a special character or not
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : ChkSpecial
//  Description   : Inputs character array coverting it into string and checks if its a special character or not
//  Input         : String
//  Output        : Boolean
//
////////////////////////////////////////////////////////////////

class Display
{
    public boolean ChkSpecial(char[] chVal, int iSize)
    {
        int iCnt = 0;
        boolean bResult = false;

        while(iCnt < iSize) 
        {
            if((chVal[iCnt] >= 'A') && (chVal[iCnt] <= 'Z') || (chVal[iCnt] >= 'a') && (chVal[iCnt] <= 'z') || (chVal[iCnt] >= '0') && (chVal[iCnt] <= '9'))
            {
                bResult = false;
            }
            else
            {
                bResult = true;
            }
            iCnt++;
        }
        return bResult;
    }    
}

public class program4
{
    public static void main(String args[])
    {
        int iCnt = 0;
        char[] cValue = null;
        String sInput = null;
        int iLength = 0;
        boolean bRet = false;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Character : ");
        sInput = sobj.nextLine();

        iLength = sInput.length();

        cValue = new char[iLength];

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            cValue[iCnt] = sInput.charAt(iCnt);
        }

        Display dobj = new Display();

        bRet = dobj.ChkSpecial(cValue, iLength);
        
        if(bRet == true) 
        {
            System.out.println("The Entered Value is a Special Character");    
        }
        else
        {
            System.out.println("The Entered Value is Not a Special Character");
        }

        sobj.close();
    }
}
